main-app/src/main/java/com/nyctransittracker/mainapp/controller/SubscriptionController.java

```diff
@@ -0,0 +1,52 @@
+package com.nyctransittracker.mainapp.controller;
+
+import com.nyctransittracker.mainapp.dto.SubscriptionRequest;
+import com.nyctransittracker.mainapp.dto.SubscriptionResponse;
+import com.nyctransittracker.mainapp.service.SubscriptionService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api/v1/subscription")
+@RequiredArgsConstructor
+public class SubscriptionController {
+
+    private final SubscriptionService service;
+
+    @PostMapping("/subscribe")
+    public ResponseEntity<SubscriptionResponse> subscribe(
+        @RequestBody SubscriptionRequest request
+    ) {
+        if (request.getUserEmail() != null && request.getStopId() != null) {
```

Comment:
you can technically return a response with a success message even if the subscription wasn't successful, like when the request.getStopId() exists but is not a valid ID and service.subscribe will still execute, just won't make any changes to DB